### ISSUE TITLE
Changing default to zsh declared Y to be an invalid choice

### ISSUE THREAD

**Issue by terrarum** [robbyrussell/oh-my-zsh#7960]
**Describe the bug**
Installed oh my zsh using the homepage curl command. When I got to `Do you want to change your default shell to zsh? [Y/n]` I typed `Y` and hit enter.

The installed then said `Invalid choice. Shell change skipped.`. 

Opening a new session has me in bash instead of zsh.

**To Reproduce**
1. Install oh-my-zsh using the homepage curl command
2. Select `Y` when asked to make zsh the default shell
3. See error

**Expected behavior**
It should accept Y as an answer to a `[Y/n]` question

**Screenshots or recordings**
![image](https://user-images.githubusercontent.com/2625755/60245501-a3bb3100-98b4-11e9-90b7-afd055ff273b.png)

**System:**
 - OS: macOS 10.14.5
 - Zsh version: 5.3
 - Terminal emulator: Terminal 2.9.5

**Comment by mcornella** [comment]
It's curious that it appears in green... Can you download the script and put an `echo $opt` after `read opt`: https://github.com/robbyrussell/oh-my-zsh/blob/5f743e1a84b9c23004bd9e4f6995939b7674d4cc/tools/install.sh#L161-L162

And also change the last line for `setup_color; setup_shell` instead of `main "$@"`: https://github.com/robbyrussell/oh-my-zsh/blob/5f743e1a84b9c23004bd9e4f6995939b7674d4cc/tools/install.sh#L269

Then run `sh install.sh` and try to reproduce the problem.

**Comment by terrarum** [comment]
I made those changes and when I run `sh install.sh` I get no output and an immediate exit.

![image](https://user-images.githubusercontent.com/2625755/60265295-50a9a400-98dd-11e9-8289-bd28c755f31f.png)

I tried it again only adding the echo and not adding the last line and it worked. I removed ohmyzsh, changed my default back to bash and ran the modified `install.sh` again:

![image](https://user-images.githubusercontent.com/2625755/60265884-bba7aa80-98de-11e9-8a05-2472f6d26d28.png)

I'll try it again unmodified and see if I get the original result.
